llvm/docs/DTLTO.rst

```diff
@@ -60,165 +80,126 @@ Distributors are programs responsible for:
-backend compilation jobs and executes the distributor passing it this file. The
-JSON file provides the following information to the distributor:
-
-- The **command line** to execute the backend compilations.
-   - DTLTO constructs a Clang command line by translating some of the LTO
-     configuration state into Clang options and forwarding options specified
-     by the user.
-
-- **Link output path**.
-   - A string identifying the output to which this LTO invocation will 
-     contribute. Distributors can use this to label build jobs for informational
-     purposes.
-
-- The list of **imports** required for each job.
-   - The per-job list of bitcode files from which importing will occur. This is
-     the same information that is emitted into import files for ThinLTO.
-
-- The **input files** required for each job.
-   - The per-job set of files required for backend compilation, such as bitcode
-     files, summary index files, and profile data.
-
-- The **output files** generated by each job.
-   - The per-job files generated by the backend compilations, such as compiled
-     object files and toolchain metrics.
-
-Temporary Files
----------------
-
-During its operation, DTLTO generates temporary files. Temporary files are
-created in the same directory as the linker's output file and their filenames
-include the stem of the bitcode module, or the output file that the LTO 
-invocation is contributing to, to aid the user in identifying them:
-
-- **JSON Job Description File**:
-    - Format:  `dtlto.<PID>.dist-file.json`
-    - Example: `dtlto.77380.dist-file.json` (for output file `dtlto.elf`).
-
-- **Object Files From Backend Compilations**:
-    - Format:  `<Module ID stem>.<Task>.<PID>.native.o`
-    - Example: `my.1.77380.native.o` (for bitcode module `my.o`).
-
-- **Summary Index Shard Files**:
-    - Format:  `<Module ID stem>.<Task>.<PID>.native.o.thinlto.bc`
-    - Example: `my.1.77380.native.o.thinlto.bc` (for bitcode module `my.o`).
-
-Temporary files are removed, by default, after the backend compilations complete.
+backend compilation jobs and executes the distributor, passing it this file.
 
 JSON Schema
 -----------
 
-Below is an example of a JSON job file for backend compilation of the module
-`dtlto.o`:
+The JSON format is explained by reference to the following example, which
+describes the backend compilation of the modules ``t1.bc`` and ``t2.bc``:
 
 .. code-block:: json
 
     {
         "common": {
             "linker_output": "dtlto.elf",
-            "linker_version": "LLD 20.0.0",
-            "args": [
-                "/usr/local/clang",
-                "-O3", "-fprofile-sample-use=my.profdata",
-                "-o", ["primary_output", 0],
-                "-c", "-x", "ir", ["primary_input", 0],
-                ["summary_index", "-fthinlto-index=", 0],
-                "--target=x86_64-sie-ps5"
-            ]
+            "args": ["clang.exe", "-c", "-flto", "-fprofile-sample-use=my.prof"],
```

Comment:
Ignoring the profiling option, is this deliberately different to the example code-gen invocations from the Bazel-like examples? Should `-ftlo` be `-flto=thin` or perhaps even be not specificed at all?

Reply:
Great comment. I'll make it as similar as possible to the Bazel-like example from earlier.  I'll remove the -flto.